File: languages/kotlin/01-basics/src/exercise_2_control_flow.kt
/*
 * Exercise 2: Control Flow
 * TODO: Implement a function that takes a score and returns a letter grade (A, B, C, D, F)
 * TODO: Implement a when expression that maps a day number (1-7) to a day name
 * TODO: Use a for loop to print numbers 1-5
 * TODO: Use a while loop to count down from 5 to 1
 */
fun getGrade(score: Int): String {
    TODO("Return A for >= 90, B for >= 80, C for >= 70, D for >= 60, F otherwise")
}

fun dayName(day: Int): String {
    TODO("Return Monday for 1, Tuesday for 2, etc., Weekend for 6/7")
}

fun main() {
    println("Grade 87: ${getGrade(87)}")
    println("Day 3: ${dayName(3)}")
    println("Count 1-5:")
    for (i in 1..5) println(i)
    println("Countdown 5-1:")
    var countdown = 5
    while (countdown > 0) {
        println(countdown)
        countdown--
    }
}

File: languages/python/01-basics/exercise_2_control_flow.py
#!/usr/bin/env python3
"""Exercise 2: Control flow — conditionals and loops."""


def grade(score: int) -> str:
    """Return a letter grade for a numeric score.

    >= 90: A, >= 80: B, >= 70: C, >= 60: D, else F.
    """
    # TODO: implement
    pass


def sum_to(n: int) -> int:
    """Return the sum of integers from 1 to n (inclusive)."""
    # TODO: implement using a for loop
    pass


def fizzbuzz(n: int) -> list[str]:
    """Return a fizzbuzz list for numbers 1 through n.

    - Multiples of 3: "fizz"
    - Multiples of 5: "buzz"
    - Multiples of both: "fizzbuzz"
    - Otherwise: the number as a string
    """
    # TODO: implement
    pass


if __name__ == "__main__":
    # Quick smoke tests — replace with real calls once implemented
    print(grade(95))
    print(sum_to(10))
    print(fizzbuzz(15))
